jetty-ee9/jetty-ee9-servlet/src/test/java/org/eclipse/jetty/ee9/servlet/DefaultServletTest.java

```diff
@@ -273,45 +307,59 @@ public void testListingFilenamesOnly() throws Exception
      * This test ensures that this behavior will not arise again.
      */
     @Test
-    @Disabled // TODO
     public void testListingFilenamesOnlyUrlResource() throws Exception
     {
-        URL extraResource = context.getClassLoader().getResource("rez/one");
-        assertNotNull(extraResource, "Must have extra jar resource in classloader");
+        Path docRoot = workDir.getEmptyPathDir().resolve("docroot");
+        FS.ensureDirExists(docRoot);
+
+        Path extraJarResources = MavenPaths.findTestResourceFile(ODD_JAR);
+        assertTrue(Files.exists(extraJarResources), "Unable to find " + ODD_JAR);
+        URL[] urls = new URL[]{extraJarResources.toUri().toURL()};
```

Comment:
Why do we need the extra classloader? The test resource will already be on the current  classpath won't it?

Reply:
This jar comes from `src/test/resources/odd_jar.jar` and isn't "live".
These test cases are specifically handling reported behaviors with odd/unusual jars.

Reply:
Since a JAR can have entries that are difficult (and in some cases impossible) on real File System (unix, mac, and windows are all included in this statement), this test case ensures that we can support these kinds of files.

Reply:
I don't really understand your answer, as all things in `src/test/resources` are always on the test classpath, but if you want to create a special classloader for it, ok.